src/sql/workbench/contrib/backup/browser/backupUiService.ts

```diff
@@ -62,21 +63,24 @@ export class BackupUiService implements IBackupUiService {
 		this._connectionUri = ConnectionUtils.generateUri(connection);
 		this._currentProvider = connection.providerName;
 		let backupDialog = this._backupDialogs[this._currentProvider];
+		const backupDialogTitle = localize('backupDialogTitle', 'Backup database - {0}:{1}', connection.serverName, connection.databaseName);
+		const backupOptions = this.getOptions(this._currentProvider);
 		if (!backupDialog) {
-			let backupOptions = this.getOptions(this._currentProvider);
 			if (backupOptions) {
 				backupDialog = this._instantiationService.createInstance(
-					OptionsDialog, 'Backup database - ' + connection.serverName + ':' + connection.databaseName, 'BackupOptions', undefined);
+					OptionsDialog, backupDialogTitle, 'BackupOptions', undefined);
 				backupDialog.onOk(() => this.handleOptionDialogClosed());
 			}
 			else {
 				backupDialog = this._instantiationService.createInstance(BackupDialog);
 			}
 			backupDialog.render();
 			this._backupDialogs[this._currentProvider] = backupDialog;
+		} else if (backupOptions) {
+			// Update the title for non-MSSQL backups each time so they show the correct database name
+			(backupDialog as OptionsDialog).title = backupDialogTitle;
```

Comment:
What happens here if backupDialog isn't an OptionsDialog?

Reply:
Do we also need to be checking the provider here like with Restore below?

Reply:
Looks like we know it's an OptionsDialog based on whether getOptions returns anything for the provider, but I'm wondering if there's some wiggle room for something to slip by in the future since we aren't checking a type or provider explicitly.

Reply:
We always create an OptionsDialog if there are options from the provider, so it was fine the way it was. (definitely not the cleanest pattern, but *shrug*. We can clean it up when we make the refactoring changes here later on).

That being said, I at least removed the cast since both dialogs inherit from modal anyways so can now update the title. The MSSQL backup/restore dialogs update the title themselves so I still only want to do this for the non-MSSQL dialogs to avoid unnecessary updates.